src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -18,6 +18,7 @@
 using AWS.Deploy.Orchestration.Data;

```

Comment:
Looking at this a bit more it seems like we have a divergence between how the CLI is determining the list and ServerMode. ServerMode is calling `GetExistingDeployedApplications` with the list of possible recommendations to filter. The CLI is calling `GetExistingDeployedApplications` without passing in a list of recommendations to filter and then calling a separate `GetCompatibleApplications` method to filter which is creating a new list and disrupting your sorting. We need to resolve this logic to have a single solution otherwise we are going to have bugs here.